src/slice.c

```diff
@@ -387,7 +387,17 @@ static SEXP lgl_as_index(SEXP i, SEXP x) {
 }
 
 static SEXP chr_as_index(SEXP i, SEXP x) {
-  SEXP nms = Rf_getAttrib(x, R_NamesSymbol);
+
+  SEXP nms = R_NilValue;
+
+  if (has_dim(x)) {
+    SEXP dim_nms = Rf_getAttrib(x, R_DimNamesSymbol);
+    if (dim_nms != R_NilValue) {
```

Comment:
I think you also need to check length is at least 1 here

Reply:
Gotcha. I've never actually seen that case before.

``` r
x <- matrix(1)
dimnames(x) <- list()
dimnames(x)
#> NULL
```